### PR TITLE
Calculate label addresses

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/parser.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ std::vector<std::string> Verifier::classB4InstructionVector {"clc", "clv", "clz"
 std::vector<std::string> tokenTypeString {"Instruction", "Number", "Lparan", "Rparan", "Colon", "Dot",
                                           "Comma", "Register", "Label", "NewLine", "Comment", "XASMEOF"};
 
-XASMParser::XASMParser(Lexer &lexer) {
+XASMParser::XASMParser(Lexer &lexer) : pc(0) {
```
Comment (rmoldo) [inline src/parser.cpp]: Please use `pc {0}` instead of `pc(0)`

File: src/parser.cpp [hunk 2]
```diff
@@ -89,6 +96,7 @@ void XASMParser::instruction() {
                         operandDest();
                         match(TokenType::Comma);
                         operandSrc();
+                        pc += 2;
```
Comment (rmoldo) [inline src/parser.cpp]: Please move `pc += 2` at the end of the function as the program counter is incremented in every case of the switch

File: src/parser.h [hunk 3]
```diff
@@ -27,6 +29,8 @@ class XASMParser {
     Lexer lexer;
     Token currentToken;
     Token nextToken;
+    std::map<std::string, uint> labels;
+    uint pc;
```
Comment (rmoldo) [inline src/parser.h]: Please use `u32` for this instead of `uint` for future consistency.

**Review by rmoldo — COMMENTED** [review]
Other than the few things I pointed out, it looks good to me

**Review by rmoldo — CHANGES_REQUESTED** [review]
Please fix the things I mentioned

File: src/parser.h (hunk 3 above)
Comment (ghost) [reply src/parser.h]: Used it to not create another dependency (include in header). But i'll go for u16 instead. Should be enough for pc. What do you think?

File: src/parser.cpp (hunk 2 above)
Comment (ghost) [reply src/parser.cpp]: Done

**Review by rmoldo — APPROVED** [review]

**Pull request merged**